packages/server/modules/workspaces/services/management.ts

```diff
@@ -323,12 +325,12 @@ export const updateWorkspaceRoleFactory =
     const workspaceRoles = await getWorkspaceRoles({ workspaceId })
     if (
       isUserLastWorkspaceAdmin(workspaceRoles, userId) &&
-      role !== Roles.Workspace.Admin
+      nextRole !== Roles.Workspace.Admin
     ) {
       throw new WorkspaceAdminRequiredError()
     }
 
-    if (role !== Roles.Workspace.Guest) {
+    if (nextRole !== Roles.Workspace.Guest) {
       const workspace = await getWorkspaceWithDomains({ id: workspaceId })
       const verifiedDomains = workspace?.domains.filter((domain) => domain?.verified)
       if (
```

Comment:
should be replaced with `userEmailsCompliantWithWorkspaceDomains` from the soon to be merged #2797 PR